lib/aryn-sdk/aryn_sdk/config.py

```diff
@@ -0,0 +1,31 @@
+from os import PathLike
+import os
+from pathlib import Path
+from typing import Optional
+import yaml
+import logging
+
+_DEFAULT_PATH = Path.home() / ".aryn" / "config.yaml"
+_ARYN_API_KEY_ENV_VAR = "ARYN_API_KEY"
+
+
+class ArynConfig:
+    def __init__(self, aryn_config_path: PathLike = _DEFAULT_PATH, aryn_api_key: Optional[str] = None):
+        self._aryn_config_path = Path(aryn_config_path)
+        self._aryn_api_key = aryn_api_key
+
+    def api_key(self) -> str:
+        if self._aryn_api_key is not None:
+            return self._aryn_api_key
+        if _ARYN_API_KEY_ENV_VAR in os.environ:
+            return os.environ[_ARYN_API_KEY_ENV_VAR]
+        if Path(self._aryn_config_path).exists():
+            with open(self._aryn_config_path, "r") as f:
+                data = yaml.safe_load(f)
+                if "aryn_token" in data:
+                    return data["aryn_token"]
+        logging.warn(
```

Comment:
Should we throw an exception here? Is there ever a case where we want to move forward without an api key?

Reply:
Maybe in some internal testing/local development of the model server it would be nice to run in less protected modes and forego this? Eh, can explode here, ok